src/xenia/kernel/objects/xthread.cc

```diff
@@ -335,12 +333,11 @@ X_STATUS XThread::Create() {
 }
 
 X_STATUS XThread::Exit(int exit_code) {
+  assert_true(XThread::GetCurrentThread() == this);
+
   // TODO(benvanik): set exit code in thread state block
 
   // TODO(benvanik); dispatch events? waiters? etc?
-  if (event_) {
```

Comment:
I thought this was needed - is it not now that the GetWaitHandle is returning the thread handle?